nat-lab/tests/mesh_api.py

```diff
@@ -353,12 +354,54 @@ def generate_peer_config(node: Node, allowed_ips: str) -> str:
                 node, ", ".join(cls.get_allowed_ip_list(node.ip_addresses))
             )
 
-        full_command = (
-            f"docker exec -d --privileged {server_config['container']} bash -c 'echo"
-            f' "{wg_conf}" > /etc/wireguard/wg0.conf; wg-quick down'
-            " /etc/wireguard/wg0.conf; wg-quick up /etc/wireguard/wg0.conf'"
-        )
-        os.system(full_command)
+            def exec_docker_commands(
+                container_id: str, commands: typing.List[str]
+            ) -> None:
+                for command in commands:
+                    cmd = f"docker exec -d --privileged {container_id} bash -c '{command}'"
+                    os.system(cmd)
```

Comment:
Does it make any sense to add this as a function instead just moving this loop to the proper `if` branch? (Because it looks that it is used only once?)

Reply:
it's not needed thus i inlined it